src/lib/cinchy.service.ts

```diff
@@ -1,31 +1,52 @@
 /* tslint:disable:max-line-length curly */
 
-import { Injectable, Inject } from '@angular/core';
+import { Observable, forkJoin, Subject, ReplaySubject, of, throwError, Subscription } from 'rxjs';
+import { map, catchError, mergeMap } from 'rxjs/operators';
+
+import { Injectable, Inject, OnDestroy } from '@angular/core';
 import { HttpClient, HttpInterceptor, HttpRequest, HttpHandler, HttpEvent, HttpHeaders } from '@angular/common/http';
+import { NavigationEnd, Router } from "@angular/router";
+
 import { OAuthService, AuthConfig, OAuthStorage } from 'angular-oauth2-oidc';
+
 import { JwksValidationHandler } from './jwks-validation-handler';
+
 import { CinchyConfig } from './cinchy.config';
 import { CinchyGlobalConfig } from './cinchy.global.config';
-import { Observable, forkJoin, Subject, ReplaySubject, of, throwError } from 'rxjs';
-import { map, catchError, mergeMap } from 'rxjs/operators';
 import { CinchyLiteralDictionary } from './cinchy.literal.dictionary';
-import { CinchyUserPreference } from './cinchy.user.preference';
 import { QueryType } from './cinchy.query.type';
+import { CinchyUserPreference } from './cinchy.user.preference';
+
 
 @Injectable({
     providedIn: 'root'
 })
-export class CinchyService {
+export class CinchyService implements OnDestroy {
 
     private cinchyRootUrl;
     private accessTokenSubject: Subject<string>;
     private userIdentitySubject: Subject<object>;
-    constructor(private _httpClient: HttpClient, private _oAuthStorge: OAuthStorage, private _oAuthService: OAuthService, private _cinchyGlobalConfig: CinchyGlobalConfig, @Inject(CinchyConfig) private config: CinchyConfig) {
+
+    private _routerSubscription: Subscription;
+
+    constructor(
+        private _httpClient: HttpClient,
+        private _oAuthStorage: OAuthStorage,
+        private _oAuthService: OAuthService,
+        private _router: Router,
+        private _cinchyGlobalConfig: CinchyGlobalConfig,
+        @Inject(CinchyConfig) private config: CinchyConfig
+    ) {
         this._cinchyGlobalConfig.setUserValues(this.config);
         this.cinchyRootUrl = this.config.cinchyRootUrl;
         this.accessTokenSubject = new ReplaySubject<string>();
         this.userIdentitySubject = new ReplaySubject<object>();
 
+
+        if (location.search.length) {
+            this._oAuthStorage.setItem("[Cinchy][login][queryParams]", location.search.substr(1));
```

Comment:
After giving it some more thought, would putting this prior to the line that calls _oAuthService.loadDiscoveryDocumentAndLogin() make more sense in the edge case where someone's app actually adjusts their query params state (without reload) prior to calling the login method? so it'll always store the latest right before login